### DISCUSSION TITLE
Emulator is throwing error while using firebase authentication with React Native using expo-cli

### DISCUSSION THREAD

**Discussion by dipanshubahuguna** [invertase/react-native-firebase#4380] [Q&A] [answered]
While using functionality of authentication through firebase, emulator is showing error i.e.{ you attempted to use a firebase module that is not installed on your android project by calling **firebase.app()** }.Please help me to resolve this error.
I'm using Expo as command line input.

react-native version -0.63.6,
"@react-native-firebase/app": "^8.4.3",
"@react-native-firebase/auth": "^9.2.3"
  
![Screenshot (159)](https://user-images.githubusercontent.com/54525722/95672841-87cec200-0bc1-11eb-9cf2-e59291da13f3.png)

**Comment by mikehardy** [comment, marked as answer]
Hi there!

There's not enough information to go on, but this will be some project-specific installation failure.
Please clone this repo and run this script to generate a fully working sample project, the difference causing your project to not run will be in the difference between the sample and your project: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

**Comment by dipanshubahuguna** [comment]
Hello @mikehardy  
I'm using expo-cli with react native and I was trying to add firebase as a backend via web configuration key.
I was trying to add  Email authentication functionality with the help of firebase but my emulator is throwing error as mentioned above . 
After a lot of research one thing I came to know is that this is a version issue as react-native ^0.63.3 is not linking all the modules but I'm not able to resolve this error from past 15 days :( .
One more thing I came to know that we can use Android/iOS config but for this we first need to split our application's file into iOS and android  using expo-split but then there will be no significance of using react-native as a cross platforming framework.
Please help me to resolve this issue I will be very thankful to you..

**Reply by mikehardy** [reply]
If you are using Expo, you'll need to work with them for support.
